Flaky io.camunda.zeebe.it.network.LargeMessageSizeTest

**Summary**

Three tests failed:

- shouldCreateInstanceWithLargeVariables
- shouldDeployLargeProcess
- shouldCompleteJobWithLargeVariables

> Try to answer the following as best as possible

- How often does the test fail? - 5
- Does it block your work?
- Do we suspect that it is a real failure?

**Failures**

> Outline known failure cases, e.g. a failed assertion and its stacktrace obtained from Jenkins

<details><summary>Example assertion failure</summary>
<pre>
shouldCreateInstanceWithLargeVariables (Flaky Test): io.camunda.zeebe.it.network.LargeMessageSizeTest
io.camunda.zeebe.client.api.command.ClientStatusException: io exception
	at io.camunda.zeebe.client.impl.ZeebeClientFutureImpl.transformExecutionException(ZeebeClientFutureImpl.java:116)
	at io.camunda.zeebe.client.impl.ZeebeClientFutureImpl.join(ZeebeClientFutureImpl.java:54)
	at io.camunda.zeebe.it.util.GrpcClientRule.deployProcess(GrpcClientRule.java:201)
	at io.camunda.zeebe.it.network.LargeMessageSizeTest.shouldCreateInstanceWithLargeVariables(LargeMessageSizeTest.java:107)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
Caused by: java.util.concurrent.ExecutionException: io.grpc.StatusRuntimeException: UNAVAILABLE: io exception
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2073)
	at io.camunda.zeebe.client.impl.ZeebeClientFutureImpl.join(ZeebeClientFutureImpl.java:52)
	... 3 more
Caused by: io.grpc.StatusRuntimeException: UNAVAILABLE: io exception
	at io.grpc.Status.asRuntimeException(Status.java:533)
	at io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onClose(ClientCalls.java:481)
	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:574)
	at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:72)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:742)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:723)
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: finishConnect(..) failed: Connection refused: /0.0.0.0:2048
Caused by: java.net.ConnectException: finishConnect(..) failed: Connection refused
	at io.netty.channel.unix.Errors.newConnectException0(Errors.java:166)
	at io.netty.channel.unix.Errors.handleConnectErrno(Errors.java:131)
	at io.netty.channel.unix.Socket.finishConnect(Socket.java:359)
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.doFinishConnect(AbstractEpollChannel.java:711)
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.finishConnect(AbstractEpollChannel.java:688)
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.epollOutReady(AbstractEpollChannel.java:567)
	at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:499)
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:407)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:1583)

 shouldDeployLargeProcess (Flaky Test): io.camunda.zeebe.it.network.LargeMessageSizeTest
io.camunda.zeebe.client.api.command.ClientStatusException: io exception
	at io.camunda.zeebe.client.impl.ZeebeClientFutureImpl.transformExecutionException(ZeebeClientFutureImpl.java:116)
	at io.camunda.zeebe.client.impl.ZeebeClientFutureImpl.join(ZeebeClientFutureImpl.java:54)
	at io.camunda.zeebe.it.network.LargeMessageSizeTest.shouldDeployLargeProcess(LargeMessageSizeTest.java:88)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
Caused by: java.util.concurrent.ExecutionException: io.grpc.StatusRuntimeException: UNAVAILABLE: io exception
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2073)
	at io.camunda.zeebe.client.impl.ZeebeClientFutureImpl.join(ZeebeClientFutureImpl.java:52)
	... 2 more
Caused by: io.grpc.StatusRuntimeException: UNAVAILABLE: io exception
	at io.grpc.Status.asRuntimeException(Status.java:533)
	at io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onClose(ClientCalls.java:481)
	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:574)
	at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:72)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:742)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:723)
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: finishConnect(..) failed: Connection refused: /0.0.0.0:2048
Caused by: java.net.ConnectException: finishConnect(..) failed: Connection refused
	at io.netty.channel.unix.Errors.newConnectException0(Errors.java:166)
	at io.netty.channel.unix.Errors.handleConnectErrno(Errors.java:131)
	at io.netty.channel.unix.Socket.finishConnect(Socket.java:359)
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.doFinishConnect(AbstractEpollChannel.java:711)
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.finishConnect(AbstractEpollChannel.java:688)
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.epollOutReady(AbstractEpollChannel.java:567)
	at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:499)
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:407)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:1583)

shouldCompleteJobWithLargeVariables (Flaky Test): io.camunda.zeebe.it.network.LargeMessageSizeTest
io.camunda.zeebe.client.api.command.ClientStatusException: io exception
	at io.camunda.zeebe.client.impl.ZeebeClientFutureImpl.transformExecutionException(ZeebeClientFutureImpl.java:116)
	at io.camunda.zeebe.client.impl.ZeebeClientFutureImpl.join(ZeebeClientFutureImpl.java:54)
	at io.camunda.zeebe.it.util.GrpcClientRule.deployProcess(GrpcClientRule.java:201)
	at io.camunda.zeebe.it.network.LargeMessageSizeTest.shouldCompleteJobWithLargeVariables(LargeMessageSizeTest.java:128)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
Caused by: java.util.concurrent.ExecutionException: io.grpc.StatusRuntimeException: UNAVAILABLE: io exception
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2073)
	at io.camunda.zeebe.client.impl.ZeebeClientFutureImpl.join(ZeebeClientFutureImpl.java:52)
	... 3 more
Caused by: io.grpc.StatusRuntimeException: UNAVAILABLE: io exception
	at io.grpc.Status.asRuntimeException(Status.java:533)
	at io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onClose(ClientCalls.java:481)
	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:574)
	at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:72)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:742)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:723)
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: finishConnect(..) failed: Connection refused: /0.0.0.0:2048
Caused by: java.net.ConnectException: finishConnect(..) failed: Connection refused
	at io.netty.channel.unix.Errors.newConnectException0(Errors.java:166)
	at io.netty.channel.unix.Errors.handleConnectErrno(Errors.java:131)
	at io.netty.channel.unix.Socket.finishConnect(Socket.java:359)
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.doFinishConnect(AbstractEpollChannel.java:711)
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.finishConnect(AbstractEpollChannel.java:688)
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.epollOutReady(AbstractEpollChannel.java:567)
	at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:499)
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:407)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:1583)
</pre>
</details>

**Hypotheses**

> List any hypotheses if you have one; can be ommitted

**Logs**

https://github.com/camunda/zeebe/actions/runs/8056857292
[Test results for [IT] QA Integration Tests.zip](https://github.com/camunda/zeebe/files/14417672/Test.results.for.IT.QA.Integration.Tests.zip)

Failed again with:

`Caused by: java.util.concurrent.CompletionException: io.netty.channel.unix.Errors$NativeIoException: bind(..) failed: Address already in use`

Test run [here](https://github.com/camunda/zeebe/actions/runs/8113457538/job/22176808835)

Exception:
```
io.camunda.zeebe.util.exception.UncheckedExecutionException: Failed to start broker
	at io.camunda.zeebe.broker.Broker.internalStart(Broker.java:111)
	at io.camunda.zeebe.util.LogUtil.doWithMDC(LogUtil.java:23)
	at io.camunda.zeebe.broker.Broker.start(Broker.java:85)
	at io.camunda.zeebe.broker.test.EmbeddedBrokerRule.startBroker(EmbeddedBrokerRule.java:248)
	at io.camunda.zeebe.broker.test.EmbeddedBrokerRule.before(EmbeddedBrokerRule.java:136)
Caused by: java.util.concurrent.ExecutionException: Startup failed in the following steps: [Cluster Services]. See suppressed exceptions for details.
	at io.camunda.zeebe.scheduler.future.CompletableActorFuture.get(CompletableActorFuture.java:147)
	at io.camunda.zeebe.scheduler.future.CompletableActorFuture.get(CompletableActorFuture.java:114)
	at io.camunda.zeebe.scheduler.FutureUtil.join(FutureUtil.java:21)
	at io.camunda.zeebe.scheduler.future.CompletableActorFuture.join(CompletableActorFuture.java:203)
	at io.camunda.zeebe.broker.Broker.internalStart(Broker.java:102)
	... 4 more
Caused by: io.camunda.zeebe.scheduler.startup.StartupProcessException: Startup failed in the following steps: [Cluster Services]. See suppressed exceptions for details.
	at io.camunda.zeebe.scheduler.startup.StartupProcess.aggregateExceptionsSynchronized(StartupProcess.java:283)
	at io.camunda.zeebe.scheduler.startup.StartupProcess.completeStartupFutureExceptionallySynchronized(StartupProcess.java:184)
	at io.camunda.zeebe.scheduler.startup.StartupProcess.lambda$proceedWithStartupSynchronized$3(StartupProcess.java:168)
	at io.camunda.zeebe.scheduler.future.FutureContinuationRunnable.run(FutureContinuationRunnable.java:33)
	at io.camunda.zeebe.scheduler.ActorJob.invoke(ActorJob.java:96)
	at io.camunda.zeebe.scheduler.ActorJob.execute(ActorJob.java:47)
	at io.camunda.zeebe.scheduler.ActorTask.execute(ActorTask.java:122)
	at io.camunda.zeebe.scheduler.ActorThread.executeCurrentTask(ActorThread.java:130)
	at io.camunda.zeebe.scheduler.ActorThread.doWork(ActorThread.java:108)
	at io.camunda.zeebe.scheduler.ActorThread.run(ActorThread.java:227)
	Suppressed: io.camunda.zeebe.scheduler.startup.StartupProcessStepException: Bootstrap step Cluster Services failed
		at io.camunda.zeebe.scheduler.startup.StartupProcess.completeStartupFutureExceptionallySynchronized(StartupProcess.java:186)
		... 8 more
	Caused by: java.util.concurrent.CompletionException: io.netty.channel.unix.Errors$NativeIoException: bind(..) failed: Address already in use
		at java.base/java.util.concurrent.CompletableFuture.encodeRelay(CompletableFuture.java:368)
		at java.base/java.util.concurrent.CompletableFuture.completeRelay(CompletableFuture.java:377)
		at java.base/java.util.concurrent.CompletableFuture$UniRelay.tryFire(CompletableFuture.java:1097)
		at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
		at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2194)
		at io.atomix.cluster.messaging.impl.NettyMessagingService.lambda$bind$41(NettyMessagingService.java:843)
		at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:590)
		at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:557)
		at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:492)
		at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:636)
		at io.netty.util.concurrent.DefaultPromise.setFailure0(DefaultPromise.java:629)
		at io.netty.util.concurrent.DefaultPromise.tryFailure(DefaultPromise.java:118)
		at io.netty.channel.AbstractChannel$AbstractUnsafe.safeSetFailure(AbstractChannel.java:999)
		at io.netty.channel.AbstractChannel$AbstractUnsafe.bind(AbstractChannel.java:564)
		at io.netty.channel.DefaultChannelPipeline$HeadContext.bind(DefaultChannelPipeline.java:1334)
		at io.netty.channel.AbstractChannelHandlerContext.invokeBind(AbstractChannelHandlerContext.java:600)
		at io.netty.channel.AbstractChannelHandlerContext.bind(AbstractChannelHandlerContext.java:579)
		at io.netty.channel.DefaultChannelPipeline.bind(DefaultChannelPipeline.java:973)
		at io.netty.channel.AbstractChannel.bind(AbstractChannel.java:260)
		at io.netty.bootstrap.AbstractBootstrap$2.run(AbstractBootstrap.java:380)
		at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:173)
		at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:166)
		at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:[470](https://github.com/camunda/zeebe/actions/runs/8113457538/job/22176808835#step:10:471))
		at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:416)
		at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
		at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
		at java.base/java.lang.Thread.run(Thread.java:1583)
	Caused by: io.netty.channel.unix.Errors$NativeIoException: bind(..) failed: Address already in use
```

Failed again ([here](https://github.com/camunda/zeebe/actions/runs/8149561630/job/22274417762)): 
```
Error:  io.camunda.zeebe.it.network.LargeMessageSizeTest.shouldCreateInstanceWithLargeVariables -- Time elapsed: 0.016 s <<< ERROR!
io.camunda.zeebe.client.api.command.ClientStatusException: io exception
	at io.camunda.zeebe.client.impl.ZeebeClientFutureImpl.transformExecutionException(ZeebeClientFutureImpl.java:116)
	at io.camunda.zeebe.client.impl.ZeebeClientFutureImpl.join(ZeebeClientFutureImpl.java:54)
	at io.camunda.zeebe.it.util.GrpcClientRule.deployProcess(GrpcClientRule.java:201)
	at io.camunda.zeebe.it.network.LargeMessageSizeTest.shouldCreateInstanceWithLargeVariables(LargeMessageSizeTest.java:107)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
Caused by: java.util.concurrent.ExecutionException: io.grpc.StatusRuntimeException: UNAVAILABLE: io exception
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2073)
	at io.camunda.zeebe.client.impl.ZeebeClientFutureImpl.join(ZeebeClientFutureImpl.java:52)
	... 3 more
Caused by: io.grpc.StatusRuntimeException: UNAVAILABLE: io exception
	at io.grpc.Status.asRuntimeException(Status.java:533)
	at io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onClose(ClientCalls.java:481)
	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:574)
	at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:72)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:742)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:723)
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1[144](https://github.com/camunda/zeebe/actions/runs/8149561630/job/22274417762#step:10:145))
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:[158](https://github.com/camunda/zeebe/actions/runs/8149561630/job/22274417762#step:10:159)3)
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: finishConnect(..) failed: Connection refused: /0.0.0.0:1248
Caused by: java.net.ConnectException: finishConnect(..) failed: Connection refused
	at io.netty.channel.unix.Errors.newConnectException0(Errors.java:[166](https://github.com/camunda/zeebe/actions/runs/8149561630/job/22274417762#step:10:167))
	at io.netty.channel.unix.Errors.handleConnectErrno(Errors.java:131)
	at io.netty.channel.unix.Socket.finishConnect(Socket.java:359)
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.doFinishConnect(AbstractEpollChannel.java:711)
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.finishConnect(AbstractEpollChannel.java:688)
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.epollOutReady(AbstractEpollChannel.java:567)
	at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:499)
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:407)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:1583)
```

--- a/zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/network/LargeMessageSizeTest.java
+++ b/zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/network/LargeMessageSizeTest.java
@@ -28,7 +28,6 @@ import java.util.Random;
 import org.assertj.core.api.Assertions;
 import org.awaitility.Awaitility;
 import org.junit.Before;
-import org.junit.ClassRule;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.RuleChain;
@@ -46,16 +45,11 @@ public final class LargeMessageSizeTest {
   private static final long METADATA_SIZE = 512;
 
   private static final String LARGE_TEXT = "x".repeat((int) (LARGE_SIZE - METADATA_SIZE));
-
-  private static final EmbeddedBrokerRule BROKER_RULE =
-      new EmbeddedBrokerRule(b -> b.getNetwork().setMaxMessageSize(MAX_MESSAGE_SIZE));
-  private static final GrpcClientRule CLIENT_RULE = new GrpcClientRule(BROKER_RULE);
-
-  @ClassRule
-  public static RuleChain ruleChain = RuleChain.outerRule(BROKER_RULE).around(CLIENT_RULE);
-
   @Rule public final BrokerClassRuleHelper helper = new BrokerClassRuleHelper();
-
+  private final EmbeddedBrokerRule brokerRule =
+      new EmbeddedBrokerRule(b -> b.getNetwork().setMaxMessageSize(MAX_MESSAGE_SIZE));
+  private final GrpcClientRule clientRule = new GrpcClientRule(brokerRule);
+  @Rule public RuleChain ruleChain = RuleChain.outerRule(brokerRule).around(clientRule);
   private String jobType;
 
   private static BpmnModelInstance process(final String jobType) {
@@ -80,7 +74,7 @@ public final class LargeMessageSizeTest {
 
     // when
     final var deployment =
-        CLIENT_RULE
+        clientRule
             .getClient()
             .newDeployResourceCommand()
             .addResourceStringUtf8(largeProcess, "process.bpmn")
@@ -91,7 +85,7 @@ public final class LargeMessageSizeTest {
 
     // then
     final var processInstanceEvent =
-        CLIENT_RULE
+        clientRule
             .getClient()
             .newCreateInstanceCommand()
             .processDefinitionKey(processDefinitionKey)
@@ -104,13 +98,13 @@ public final class LargeMessageSizeTest {
   @Test
   public void shouldCreateInstanceWithLargeVariables() {
     // given
-    final var processDefinitionKey = CLIENT_RULE.deployProcess(process(jobType));
+    final var processDefinitionKey = clientRule.deployProcess(process(jobType));
 
     // when
     final Map<String, Object> largeVariables = Map.of("largeVariable", LARGE_TEXT);
 
     final var processInstanceEvent =
-        CLIENT_RULE
+        clientRule
             .getClient()
             .newCreateInstanceCommand()
             .processDefinitionKey(processDefinitionKey)
@@ -125,10 +119,10 @@ public final class LargeMessageSizeTest {
   @Test
   public void shouldCompleteJobWithLargeVariables() {
     // given
-    final var processDefinitionKey = CLIENT_RULE.deployProcess(process(jobType));
+    final var processDefinitionKey = clientRule.deployProcess(process(jobType));
 
     final var processInstanceEvent =
-        CLIENT_RULE
+        clientRule
             .getClient()
             .newCreateInstanceCommand()
             .processDefinitionKey(processDefinitionKey)
@@ -138,7 +132,7 @@ public final class LargeMessageSizeTest {
     // when
     final Map<String, Object> largeVariables = Map.of("largeVariable", LARGE_TEXT);
 
-    CLIENT_RULE
+    clientRule
         .getClient()
         .newWorker()
         .jobType(jobType)
@@ -162,7 +156,7 @@ public final class LargeMessageSizeTest {
             .endEvent()
             .done();
 
-    CLIENT_RULE
+    clientRule
         .getClient()
         .newDeployResourceCommand()
         .addProcessModel(modelInstance, "foo.bpmn")
@@ -175,7 +169,7 @@ public final class LargeMessageSizeTest {
 
     final int numberOfJobsToActivate = 5;
     for (int i = 0; i < numberOfJobsToActivate; i++) {
-      CLIENT_RULE
+      clientRule
           .getClient()
           .newCreateInstanceCommand()
           .bpmnProcessId("foo")
@@ -193,7 +187,7 @@ public final class LargeMessageSizeTest {
 
     // when
     final JobWorkerBuilderStep3 builder =
-        CLIENT_RULE.getClient().newWorker().jobType("foo").handler(COMPLETING_JOB_HANDLER);
+        clientRule.getClient().newWorker().jobType("foo").handler(COMPLETING_JOB_HANDLER);
 
     // then
     try (final JobWorker ignored = builder.open()) {
@@ -222,8 +216,8 @@ public final class LargeMessageSizeTest {
     final DataSize maxMessageSize = DataSize.ofBytes(5100);
     final int jobVariableSize = 144;
 
-    BROKER_RULE.getBrokerCfg().getGateway().getNetwork().setMaxMessageSize(maxMessageSize);
-    BROKER_RULE.restartBroker();
+    brokerRule.getBrokerCfg().getGateway().getNetwork().setMaxMessageSize(maxMessageSize);
+    brokerRule.restartBroker();
 
     // given
     final var modelInstance =
@@ -234,7 +228,7 @@ public final class LargeMessageSizeTest {
             .endEvent()
             .done();
 
-    CLIENT_RULE
+    clientRule
         .getClient()
         .newDeployResourceCommand()
         .addProcessModel(modelInstance, "foo.bpmn")
@@ -246,7 +240,7 @@ public final class LargeMessageSizeTest {
 
     final int numberOfJobsToActivate = 5;
     for (int i = 0; i < numberOfJobsToActivate; i++) {
-      CLIENT_RULE
+      clientRule
           .getClient()
           .newCreateInstanceCommand()
           .bpmnProcessId("foo")
@@ -264,7 +258,7 @@ public final class LargeMessageSizeTest {
 
     // when
     final JobWorkerBuilderStep3 builder =
-        CLIENT_RULE.getClient().newWorker().jobType("foo").handler(COMPLETING_JOB_HANDLER);
+        clientRule.getClient().newWorker().jobType("foo").handler(COMPLETING_JOB_HANDLER);
 
     // then
     try (final JobWorker ignored = builder.open()) {
@@ -282,7 +276,7 @@ public final class LargeMessageSizeTest {
     }
 
     // reset max message size to the initial value
-    BROKER_RULE.getBrokerCfg().getGateway().getNetwork().setMaxMessageSize(MAX_MESSAGE_SIZE);
-    BROKER_RULE.restartBroker();
+    brokerRule.getBrokerCfg().getGateway().getNetwork().setMaxMessageSize(MAX_MESSAGE_SIZE);
+    brokerRule.restartBroker();
   }
 }

